### PR TITLE
Update to Kafka 2.0.0

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- ~~Bugfix~~
- Enhancement / new feature
- ~~Refactoring~~

### Description

Update Kafka to 2.0.0 - **Work In Progress using RC2**.

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging

### CODE REVIEW

**Comment by strimzici** [comment]
Passed for Jenkins

**Comment by seglo** [comment]
2.0.0 GA just dropped today. 

https://lists.apache.org/thread.html/ba8fc98076fbbdcb5c125ece6c92e90bf6695794410752876f084bb4@%3Cusers.kafka.apache.org%3E

**Comment by scholzj** [comment]
Rebased this PR to the final release of Kafka 2.0.0. t should be ready for review now.

File: Makefile.docker [hunk 1]
```diff
@@ -16,7 +16,7 @@ all: docker_build docker_push
 
 docker_build:
 	# Build Docker image ...
-	docker build -q --build-arg strimzi_version=$(RELEASE_VERSION) -t strimzi/$(PROJECT_NAME):latest $(DOCKERFILE_DIR)
+	docker build -q $(DOCKER_BUILD_ARGS) --build-arg strimzi_version=$(RELEASE_VERSION) -t strimzi/$(PROJECT_NAME):latest $(DOCKERFILE_DIR)
```
Comment (ppatierno) [inline Makefile.docker]: is this `DOCKER_BUILD_ARGS` something new?

Comment (scholzj) [reply Makefile.docker]: Yeah, I think I added because it is useful when you need to for rebuild (`--no-cache`).

**Pull request merged**